src/interactivity/win32/ut_interactivity_win32/UiaTextRangeTests.cpp

```diff
@@ -1136,57 +1136,82 @@ class UiaTextRangeTests
+        VERIFY_SUCCEEDED(TestData::TryGetValue(L"degenerate", degenerate), L"Get degenerate variant");
+        TextUnit textUnit{ static_cast<TextUnit>(unit) };
+
         Microsoft::WRL::ComPtr<UiaTextRange> utr;
         int moveAmt;
-        for (int unit = TextUnit::TextUnit_Character; unit != TextUnit::TextUnit_Document; ++unit)
-        {
-            Log::Comment(NoThrowString().Format(L"Forward by %s", toString(static_cast<TextUnit>(unit))));
+        Log::Comment(NoThrowString().Format(L"Forward by %s", toString(textUnit)));
 
-            // Create a degenerate UTR at EndExclusive
+        // Create an UTR at EndExclusive
+        if (degenerate)
+        {
+            THROW_IF_FAILED(Microsoft::WRL::MakeAndInitialize<UiaTextRange>(&utr, _pUiaData, &_dummyProvider, endExclusive, endExclusive));
+        }
+        else
+        {
             THROW_IF_FAILED(Microsoft::WRL::MakeAndInitialize<UiaTextRange>(&utr, _pUiaData, &_dummyProvider, endInclusive, endExclusive));
-            THROW_IF_FAILED(utr->Move(static_cast<TextUnit>(unit), 1, &moveAmt));
-
-            VERIFY_ARE_EQUAL(endExclusive, utr->_end);
-            VERIFY_ARE_EQUAL(0, moveAmt);
         }
+        THROW_IF_FAILED(utr->Move(textUnit, 1, &moveAmt));
+
+        VERIFY_ARE_EQUAL(endExclusive, utr->_end);
+        VERIFY_ARE_EQUAL(0, moveAmt);
 
         // Verify that moving backwards still works properly
+
+        // write "temp" at (2,2)
         const COORD writeTarget{ 2, 2 };
         _pTextBuffer->Write({ L"temp" }, writeTarget);
-        for (int unit = TextUnit::TextUnit_Character; unit != TextUnit::TextUnit_Document; ++unit)
-        {
-            COORD expectedEnd;
-            switch (static_cast<TextUnit>(unit))
-            {
-            case TextUnit::TextUnit_Character:
-                expectedEnd = endInclusive;
-                break;
-            case TextUnit::TextUnit_Word:
-                expectedEnd = writeTarget;
-                break;
-            case TextUnit::TextUnit_Line:
-                expectedEnd = endExclusive;
-                break;
-            case TextUnit::TextUnit_Document:
-                expectedEnd = bufferSize.Origin();
-                break;
-            default:
-                continue;
-            }
 
-            Log::Comment(NoThrowString().Format(L"Backwards by %s", toString(static_cast<TextUnit>(unit))));
+        //if (textUnit == TextUnit::TextUnit_Character && !degenerate)
+        //    DebugBreak();
 
-            // Create a degenerate UTR at EndExclusive
-            THROW_IF_FAILED(Microsoft::WRL::MakeAndInitialize<UiaTextRange>(&utr, _pUiaData, &_dummyProvider, endInclusive, endExclusive));
-            THROW_IF_FAILED(utr->Move(static_cast<TextUnit>(unit), -1, &moveAmt));
+        Log::Comment(NoThrowString().Format(L"Backwards by %s", toString(textUnit)));
+        THROW_IF_FAILED(utr->Move(textUnit, -1, &moveAmt));

```

Comment:
so this test is different from before. instead of making a new UTR, it assumes that the existing one was ... unchanged? or it builds off the existing one?

This could use a comment explaining why we're recycling the UTR and what its expected state is!

Reply:
nevermind -- now I understand. You're ensuring that the range _goes degenerate off the end_ no matter what.

Reply:
Correct. We already verified that the UTR failed to move in the first place. So rather than creating another UTR at the same spot, we can save a bit of time.